ui/src/path.ts

```diff
@@ -22,6 +27,58 @@ export const projectSourceCommit = (urn: Urn, hash: string): string =>
   `/projects/${urn}/source/commit/${hash}`;
 export const projectSourceCommits = (urn: Urn): string =>
   `/projects/${urn}/source/commits`;
+
+export enum PatchFilter {
+  Open = "open",
+  Closed = "closed",
+  All = "all",
+}
+
+export interface PatchesQuery {
+  filter: PatchFilter;
+}
+
+const patchesQuerySchema = zod
+  .object({
+    filter: zod.enum([PatchFilter.Open, PatchFilter.Closed, PatchFilter.All]),
+  })
+  .partial();
+
+export const projectSourcePatchesFilter = (
+  urn: Urn,
+  filter: PatchFilter
+): string => `${projectSourcePatches(urn)}?${qs.stringify({ filter })}`;
+
+export const projectSourcePatches = (urn: Urn): string =>
+  `/projects/${urn}/source/patches`;
+
+// Holds the query paramters for the patches route if that route is
+// active. Otherwise holds `undefined`
+export const projectSourcePatchesQuery = (): svelteStore.Readable<
```

Comment:
The way filtering is implemented here and in the respective component is quite elegant! 👍